File: InventoryManagementServices/ReactApp/inventory-management-react/src/app/models/movieSearch.ts
export interface MovieSearch
{
    title?: string;
    description?: string;
    releaseYear?: string;
    movieRatingsId?: string;
    movieGenreIds?: [];
}


export class MovieSearch implements MovieSearch {
    constructor(init?: MovieSearchFormValues) {
        Object.assign(this, init);
    }
}

  export class MovieSearchFormValues {
    title: string = '';
    description: string= '';
    releaseYear: string= '';
    movieRatingsId: string= '';
    movieGenreIds?: [] = undefined;

    constructor(game?: MovieSearchFormValues) {
      if (game) {
        this.title = game.title;
        this.description = game.description;
        this.releaseYear = game.releaseYear;
        this.movieRatingsId = game.movieRatingsId;
        this.movieGenreIds = game.movieGenreIds;
      }
    }
  }

File: InventoryManagementServices/Services/Movie/Movie.Infrastructure/Entities/MovieSearch.cs
namespace Movie.Infrastructure.Entities
{
    public class MovieSearch
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public int? ReleaseYear { get; set; }
        public Guid? MovieRatingsId { get; set; }
        public List<int> MovieGenreIds { get; set; }
    }
}
